### PR TITLE
Services and controllers

### PR DESCRIPTION
**Pull request by DorWolfgorAcademy**
Chef and Restaurant are done.

### CODE REVIEW

File: src/main/java/com/controller/ChefController.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+public class ChefController {
+    ChefService chefService;
+
+    @Autowired
+    public ChefController(ChefService chefService) {
+        this.chefService = chefService;
+    }
+
+    @GetMapping("/chefs")
+    List<Chef> all(@RequestParam(name = "sort-by") String sortBy) {
```
Comment (omrip2) [inline src/main/java/com/controller/ChefController.java:20]: The sort by can be an enum

File: src/main/java/com/controller/HomeController.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package com.controller;
+
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class HomeController {
+    @RequestMapping(value = {"/", "/homepage"})
+    public String  getHomePage(){
+        return "home";
```
Comment (omrip2) [inline src/main/java/com/controller/HomeController.java:10]: Add TODO comment for all the unimplemented endpoints so you won't forget to implement it
The opposite from forgetting is writing

File: src/main/java/com/controller/RestaurantController.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+public class RestaurantController {
+    RestaurantService restaurantService;
+
+   @Autowired
+    public RestaurantController(RestaurantService restaurantService) {
```
Comment (omrip2) [inline src/main/java/com/controller/RestaurantController.java:15]: Add validations to all endpoints

File: src/main/java/com/controller/RestaurantController.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        this.restaurantService = restaurantService;
+    }
+
+    @GetMapping("/restaurants")
+    List<Restaurant> all(@RequestParam(required = false) Double latitude,
+                         @RequestParam(required = false) Double longitude,
```
Comment (omrip2) [inline src/main/java/com/controller/RestaurantController.java:21]: Add default values to all params

File: src/main/java/com/service/RestaurantService.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        if(distance!=null){
+            if(latitude == null || longitude == null){
+                throw new IllegalArgumentException("Please enter your location");
+            }
+        }
+        if((minDishPrice==null&&maxDishPrice!=null)||(minDishPrice!=null&&maxDishPrice==null)){
+            throw new IllegalArgumentException("Please enter minimum dish price and maximum dish price values together");
+        }
+        List<Restaurant> restaurants= restaurantRepository.findAll();
+        restaurants = filterRestaurants(restaurants,minDishPrice,maxDishPrice,rating,distance,latitude,longitude);
```
Comment (omrip2) [inline src/main/java/com/service/RestaurantService.java:37]: The filter should be in the db and not in the service

**Review by omrip2 — CHANGES_REQUESTED** [review]

File: src/main/java/com/controller/HomeController.java (hunk 2 above)
Comment (DorWolfgorAcademy) [reply src/main/java/com/controller/HomeController.java:10]: fixed

File: src/main/java/com/controller/ChefController.java (hunk 1 above)
Comment (DorWolfgorAcademy) [reply src/main/java/com/controller/ChefController.java:20]: fixed

File: src/main/java/com/service/RestaurantService.java (hunk 5 above)
Comment (DorWolfgorAcademy) [reply src/main/java/com/service/RestaurantService.java:37]: fixed

File: src/main/java/com/controller/RestaurantController.java (hunk 4 above)
Comment (DorWolfgorAcademy) [reply src/main/java/com/controller/RestaurantController.java:21]: fixed

File: src/main/java/com/controller/RestaurantController.java (hunk 3 above)
Comment (DorWolfgorAcademy) [reply src/main/java/com/controller/RestaurantController.java:15]: fixed

**Review by omrip2 — APPROVED** [review]

**Pull request merged**